test/unit/firebase-app.spec.ts

```diff
@@ -149,6 +157,97 @@ describe('FirebaseApp', () => {
+
+    it('should throw when the environment variable points to an empty file', () => {
+      process.env[FIREBASE_CONFIG_FILE_VAR] = './test/resources/firebase_config_empty.json';
+      expect(() => {
+        firebaseNamespace.initializeApp();
+      }).to.throw(`Failed to parse app options file`);
+    });
+
+    it('should throw when the environment variable points to bad json', () => {
+      process.env[FIREBASE_CONFIG_FILE_VAR] = './test/resources/firebase_config_bad.json';
+      expect(() => {
+        firebaseNamespace.initializeApp();
+      }).to.throw(`Failed to parse app options file`);
+    });
+
+    it('should ignore a bad config key', () => {
+      process.env[FIREBASE_CONFIG_FILE_VAR] = './test/resources/firebase_config_bad_key.json';
+      const app = firebaseNamespace.initializeApp();
+      expect(app.options.projectId).to.equal("hipster-chat-mock");
+      expect(app.options.databaseAuthVariableOverride).to.be.undefined;
+      expect(app.options.databaseURL).to.undefined;
+      expect(app.options.storageBucket).to.undefined;
+    });
+
+    it('should ignore when the config file has a bad key, unused', () => {
+      process.env[FIREBASE_CONFIG_FILE_VAR] = './test/resources/firebase_config_bad_key.json';
+      const app = firebaseNamespace.initializeApp(mocks.appOptionsWithOverride, mocks.appName);
+      expect(app.options.projectId).to.equal("project_id");
+      expect(app.options.databaseAuthVariableOverride).to.deep.equal({ 'some#string': 'some#val' });
+      expect(app.options.databaseURL).to.equal('https://databaseName.firebaseio.com');
+      expect(app.options.storageBucket).to.equal('bucketName.appspot.com');
+    });
+
+    it('should use explicitly specified options when available and ignore the config file', () => {
+      process.env[FIREBASE_CONFIG_FILE_VAR] = './test/resources/firebase_config.json';
+      const app = firebaseNamespace.initializeApp(mocks.appOptions, mocks.appName);
+      expect(app.options.credential).to.be.instanceOf(CertCredential);
+      expect(app.options.databaseAuthVariableOverride).to.be.undefined;
+      expect(app.options.databaseURL).to.equal('https://databaseName.firebaseio.com');
+      expect(app.options.projectId).to.be.undefined;
+      expect(app.options.storageBucket).to.equal('bucketName.appspot.com');
+    });
+
+    it('should ignore the config file even if some fields are missing', () => {
+      process.env[FIREBASE_CONFIG_FILE_VAR] = './test/resources/firebase_config_partial.json';
+      const app = firebaseNamespace.initializeApp(mocks.appOptionsAuthDB, mocks.appName);
+      expect(app.options.databaseURL).to.equal('https://databaseName.firebaseio.com');
+      expect(app.options.projectId).to.be.undefined;
+      expect(app.options.storageBucket).to.be.undefined;
+    });
+
+    it('should not throw when the config environment variable is not set', () => {
+      const app = firebaseNamespace.initializeApp(mocks.appOptionsNoDatabaseUrl, mocks.appName);
```

Comment:
No options should be passed here -- I think.

Reply:
that is the following case.
(this case can be deleted, but no harm in leaving it in)